react/content-list.jsx

```diff
@@ -45,14 +58,30 @@ let ContentList = React.createClass({
     return this.setStateFromDattCore()
   },
 
+  componentWillUnmount: function () {
+    this.unmonitorDattCore()
+  },
+
   componentWillReceiveProps: function () {
     return this.setStateFromDattCore()
   },
 
   monitorDattCore: function () {
     let dattcore = this.props.dattcore
+    let uiEvents = this.props.uiEvents
+
     dattcore.on('peers-content-auth', this.handlePeersContentAuth)
     dattcore.on('content-content-auth', this.handleContentContentAuth)
+    uiEvents.on('refresh-content', this.handleRefreshContent)
+  },
+
+  unmonitorDattCore: function () {
```

Comment:
Good question. It is called in componentWillUnmount to remove event listeners that refer to the unmounted component. 
Not removing event listeners causes setState warnings like the following:
```
Warning: setState(...): Can only update a mounted or mounting component. This usually means you called setState() on an unmounted component. This is a no-op.
```
These still pop up in the logs but a lot less after removing content-list's dattcore listeners whenever it is unmounted via "unmonitorDattCore"